src/App.tsx

```diff
@@ -14,11 +14,96 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[];
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  setSelectedGoods = (goods: string) => {
+    this.setState(((prevstate: State) => {
+      return { selectedGoods: [...prevstate.selectedGoods, goods] };
+    }));
+  };
+
+  removeSelectedGood = (good: string) => {
+    this.setState((prevState => ({
+      selectedGoods: prevState.selectedGoods.filter(remaining => (
```

Comment:
be consistent, in the first one you specify `prevstate: State`, but in the second one simply `prevstate`. decide which one you prefer and be consistent